packages/superset-ui-query/src/buildQueryObject.ts

```diff
@@ -42,7 +42,7 @@ export default function buildQueryObject<T extends QueryFormData>(formData: T):
     metrics: processMetrics(formData),
     order_desc: typeof order_desc === 'undefined' ? true : order_desc,
     orderby: [],
-    row_limit: Number(row_limit),
+    row_limit: !Number(row_limit) && row_limit !== 0 ? undefined : Number(row_limit),
```

Comment:
if we're converting `row_limit` to a `number`, won't `row_limit !== 0` _always_ be `false`?

could this be something like 
```ts
const numericRowLimit = Number(row_limit);
...
row_limit: row_limit == null || isNaN(numericRowLimit) ? undefined : numericRowLimit,
```

Reply:
I'm catching the case where if row_limit = 0, we don't want to set it to undefined.

`!Number(0) is true`, but we don't want row_limit to be undefined

Reply:
maybe this should be Number(row_limit) though...

Reply:
i see your point, updated